### ISSUE TITLE
[wikipedia] Queries with e.g. & do not work properly

### ISSUE THREAD

**Issue by dgw** [sopel-irc/sopel#1255]
````
<dgw> ;w Miraculous Ladybug
<Kaede> KeyError: u'extract' (file "/usr/local/lib/python2.7/dist-packages/sopel/modules/wikipedia.py",
        line 87, in mw_snippet)
[…]
<dgw> https://en.wikipedia.org/wiki/Miraculous:_Tales_of_Ladybug_%26_Cat_Noir
<Kaede> KeyError: u'extract' (file "/usr/local/lib/python2.7/dist-packages/sopel/modules/wikipedia.py",
        line 87, in mw_snippet)
````

Basically, if a search returns a page title that contains `&`, e.g. "Miraculous: Tales of Ladybug & Cat Noir", the `wikipedia` module winds up trying to retrieve a snippet for, e.g., "Miraculous: Tales of Ladybug " and failing to do so because the page doesn't exist.

I have a fix for this in testing, but wanted to open the issue so anyone searching for the problem would find out that it is already on the radar.

**Comment by kwaaak** [comment]
https://github.com/sopel-irc/sopel/pull/1252/commits/17c44c08aa971a1b787964e07792ea36919a0982

**Comment by dgw** [comment]
@kwaaak Does aa2c694 cover both cases from #1252? It should. I've been running it for some time and haven't seen any issues. (That said, I'm now testing instead doing the `quote` in `say_snippet()` instead of `mw_snippet()`. Seems to work more or less identically, just safer if `show_url` is true because the bot then prints a safely encoded URL.)

**Comment by kwaaak** [comment]
Yes, works, at least for Python3
Example URL with a + for completeness sake : https://en.wikipedia.org/wiki/PSR_B1257+12

**Comment by dgw** [comment]
This will definitely be fixed in 6.5.2. Thanks for testing, @kwaaak!
